url3aa01d34ec_katsuster_ememu_tgz-pJ8-net_katsuster_ememu_ui_MainJ8/src/net/katsuster/ememu/ui/ARMLinuxLoader.java

```diff
@@ -0,0 +1,217 @@
+package net.katsuster.ememu.ui;
+
+import java.io.*;
+import java.net.*;
+
+import net.katsuster.ememu.generic.*;
+import net.katsuster.ememu.arm.core.ARMv5;
+
+/**
+ * ARM Linux 用の簡易ブートローダです。
+ *
+ * @author katsuhiro
+ */
+public class ARMLinuxLoader {
+    public static final int ATAG_NONE      = 0x00000000;
+    public static final int ATAG_CORE      = 0x54410001;
+    public static final int ATAG_MEM       = 0x54410002;
+    public static final int ATAG_VIDEOTEXT = 0x54410003;
+    public static final int ATAG_RAMDISK   = 0x54410004;
+    public static final int ATAG_INITRD2   = 0x54420005;
+    public static final int ATAG_SERIAL    = 0x54410006;
+    public static final int ATAG_REVISION  = 0x54410007;
+    public static final int ATAG_VIDEOLFB  = 0x54410008;
+    public static final int ATAG_CMDLINE   = 0x54410009;
+
+    public static int loadURIResource(URI uri, CPU cpu, int addr) {
+        int i;
+
+        System.out.println("loadURL: " + uri.toString());
+
+        try {
+            DataInputStream s = new DataInputStream(
+                    new BufferedInputStream(uri.toURL().openStream()));
+
+            i = 0;
+            try {
+                while (true) {
+                    cpu.write8_a32(addr + i, s.readByte());
+                    i++;
+                }
+            } catch (EOFException e) {
+                //end
+            }
+        } catch (IOException e) {
+            e.printStackTrace(System.err);
+            throw new IllegalArgumentException(e);
+        }
+
+        System.out.printf("loadURL: '%s' done, %dbytes.\n",
+                uri.toString(), i);
+
+        return i;
+    }
+
+    public static void bootFromURI(ARMv5 cpu, RAM ramMain, String kimage, String initrd, String cmdline) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 59 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.